zerver/views/webhooks/greenhouse.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import absolute_import
+
+from django.utils.translation import ugettext as _
+from django.http import HttpRequest, HttpResponse
+from typing import Any
+
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+from zerver.models import UserProfile, Client
+
+import ujson
+
+def message_creator(action, application):
+    # type: (str, Dict[str, Any]) -> str
+    def list_format(some_list):
+        # type: (List[Any]) -> str
+        internal_template = ''
+        for item in some_list:
+            item_type = item.get('type').title()
+            item_value = item.get('value')
+            item_url = item.get('url')
+            if item_type and item_value:
+                internal_template += "{}\n{}\n".format(item_type, item_value)
+            elif item_type and item_url:
+                internal_template += "[{}]({})\n".format(item_type, item_url)
+        return internal_template
+    message = ("Applying for role:" +
```

Comment:
Could we make a template constant above all functions:
`MESSAGE_TEMPLATE = "Applying for role:\n{}...."
`message  = MESSAGE_TEMPLATE.format(...)